reframe/core/buildsystems.py

```diff
@@ -481,6 +481,15 @@ class ConfigureBasedBuildSystem(BuildSystem):
     #: :type: :class:`str`
     #: :default: :class:`None`
     srcdir = variable(str, type(None), value=None)
+    
+    #: The directory of the configure script.
+    #:
+    #: This can be changed to do an out of source build without copying the
+    #: entire source tree.
+    #:
+    #: :type: :class:`str`
+    #: :default: ``'.'``
```

Comment:
Add a `.. versionchanged:: 4.8` annotation stating that this variable is now available in all configure-based build systems (Autotools, CMake).